components/global/core/io.cellery.observability.api/src/main/java/io/cellery/observability/api/auth/OIDCOauthManager.java

```diff
@@ -62,17 +63,27 @@
     private char[] clientSecret;
 
     public String getClientId() {
-        return clientId;
+        if (clientId != null) {
+            return clientId;
+        } else {
+            ServiceHolder.setOidcOauthManager(new OIDCOauthManager());
```

Comment:
Simply call the get getClientCredentials() method and set the values here. No need to re-instantiate the OIDC OAuth Manager

Reply:
Fixed as per above comment.